packages/preact/src/index.ts

```diff
@@ -80,8 +80,13 @@ function Text(this: AugmentedComponent, { data }: { data: Signal }) {
 			}
 		}
 
-		// Replace this component's vdom updater with a direct text one:
 		this._updater!._callback = () => {
+			if (isValidElement(s.peek()) || this.base?.nodeType !== 3) {
+				this._updateFlags |= HAS_PENDING_UPDATE;
+				this.setState({});
+				return;
+			}
+
 			(this.base as Text).data = s.peek();
 		};
```

Comment:
Wouldn't this be equivalent and maybe slightly easier to read, this is what I meant in my comment

```suggestion
        if (this.base?.nodeType === 3 || !isValidElement(s.peek())) {
        	this._updater!._callback = () => {
				(this.base as Text).data = s.peek();
		 	};
        }

```

Reply:
No, this code throws, because: We cannot read `s` value before initializing.
And it will not work the same way, because:
1. While first rendering this.base will never exist, 
2. this.base.nodeType and s.peek() can change while lifetime of component as well